apps/provider-console/src/components/dashboard/FinanceCard.tsx

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from "react";
+import { Card, CardContent, Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@akashnetwork/ui/components";
+
+interface PercentChangeProps {
+  currentPrice: number | null;
+  previousPrice: number | null;
+}
+
+interface FinanceCardProps {
+  title: string;
+  subtitle: string;
+  currentPrice: number | null;
+  previousPrice: number | null;
+  message: string | null;
+}
+
+const PercentChange: React.FC<PercentChangeProps> = ({ currentPrice, previousPrice }) => {
+  const { percentageChange, formattedChange } = useMemo(() => {
+    if (currentPrice === null || previousPrice === null || previousPrice === 0) {
+      return { percentageChange: 0, formattedChange: "0" };
+    }
+
+    const percentageChange = ((currentPrice - previousPrice) / previousPrice) * 100;
+    const formattedChange = Math.abs(percentageChange).toFixed(2);
+
+    return { percentageChange, formattedChange };
+  }, [currentPrice, previousPrice]);
+
+  if (currentPrice === null || previousPrice === null || previousPrice === 0) {
+    return <span className="text-gray-500">0%</span>;
+  }
+
+  if (percentageChange > 0) {
+    return <span className="text-green-500">+{formattedChange}%</span>;
+  } else if (percentageChange < 0) {
+    return <span className="text-red-500">-{formattedChange}%</span>;
+  } else {
+    return <span className="text-gray-500">0%</span>;
+  }
```

Comment:
Probably cleaner to do something like this:
```suggestion
  const isZero = currentPrice === null || previousPrice === null || previousPrice === 0 || percentageChange === 0
  const value = percentageChange !== 0 ? formattedChange : 0
  const prefix = !isZero && percentageChange > 0 ? "+" : "-"
  
  return (
    <span className={cn({ "text-gray-500": isZero, "text-green-500": percentageChange > 0, "text-red-500": percentageChange < 0 })}>
      {prefix}{value}%
    </span>
  )
```

Reply:
fixed and used your version of cleaner code